subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -224,6 +277,211 @@ static int validate_ak_order(void)
 	return 0;
 }
 
+static int bond_name_gen(char *name, uint8_t index)
+{
+	int n;
+
+	n = snprintf(name, SETTINGS_BOND_NAME_MAX_SIZE, "%s%u", SETTINGS_BOND_FULL_PREFIX, index);
+	__ASSERT_NO_MSG(n < SETTINGS_BOND_NAME_MAX_SIZE);
+	if (n < 0) {
+		return n;
+	}
+
+	return 0;
+}
+
+static size_t fp_bond_idx_get(struct fp_bond *bond)
+{
+	return bond - fp_bonds;
+}
+
+static int fp_bond_settings_delete(struct fp_bond *bond)
```

Comment:
Function naming is getting a bit counterintuitive here. Maybe we can create one function instead of `fp_bond_settings_delete`, `fp_bond_delete` and `bond_remove_completely`. Separating `fp_bond_settings_delete` may not be necessary. The new function could take `bool ble_stack_bond_remove` as an argument to decide if we need to remove bonding information from BLE stack).

Reply:
Maybe `fp_bond_info_settings_delete(struct fp_bond_info *bond)` and `bond_remove(struct fp_bond_info *bond, bool bt_stack_bond_remove)`?